rfcs/45-stageless.md

```diff
@@ -2,488 +2,457 @@
+        .add_system(spawn_more_enemies.run_if(|difficulty: Res<Difficulty>| difficulty >= 9000))
+        // The `resource_exists` and `resource_equals` helper functions for generating
+        // simple conditions (well, closures that convert into conditions).
+        .add_system(gravity.run_if(resource_exists(Gravity)))
+        // Conditions can be attached to system sets, so you can skip the whole set.
+        .add_set(GameSet::Physics.run_if(resource_equals(Paused(false))))
+        .run();
 }
 ```
 
-There are a few important subtleties to bear in mind when working with run criteria:
+### Running scheduled systems
 
-- when multiple run criteria are attached to the same system, the system will run if and only if all of those run criteria return true
-- each run criteria are evaluated once during each pass of the schedule
-  - this occurs just before the first system controlled by that criteria would be run
-  - the value is then shared for all other systems with that run criteria, ensuring all systems in the set stay synchronized
+The systems, sets, and scheduling data added to an `App` is stored on the `World` in the `SystemRegistry` resource. This resource is also responsible for generating and maintaining the dependency graphs for the stored system sets.
 
-This is important to ensure that the state of the world always matches the state expected by the system at the time it is executed.
+In order to run a system or system set on the `World`, the system(s) must be extracted from the `SystemRegistry`. (This lets the `SystemRegistry` remain accessible by the running systems).
 
-#### States
+To do that, the `SystemRegistry` has methods to export a system or schedule given a label. A **schedule** is an executable version of a system set, containing all the systems and conditions, as well as some metadata for an **executor** to run those in the correct order (efficiently). After the schedule has been run, it can be returned to the `SystemRegistry` using another method. This is the same process the default `App` runner uses to execute the startup sequence and main update loop.
 
-**States** allow you to toggle systems on-and-off based on the value of a given resource and smoothly handle transitions between states by running cleanup and initialization systems.
-Typically, these are used for relatively complex, far-reaching facets of the game like pausing, loading assets or entering into menus.
+### Command processing
 
-The current value of each state is stored in a resource, which must derive the `State` trait.
-These are typically (but not necessarily) enums, where each distinct state is represented as an enum variant.
+Commands are arbitrary world modifications that are most commonly used to spawn or despawn entities and add or remove components. Since those types of changes have unpredictable side effects on stored component data, commands are deferred until the next scheduled `apply_buffers` exclusive system runs.

```

Comment:
```suggestion
Commands are arbitrary world modifications that are most commonly used to spawn or despawn entities and add or remove components.
Since those types of changes have unpredictable side effects on stored component data, commands are deferred until the next scheduled `apply_buffers` exclusive system runs.
```